reference/common/models/highalert.v1.yaml

```diff
@@ -0,0 +1,12 @@
+type: object
```

Comment:
It may be good to define a common base type `alert` that contains all the common fields:

```
enabled:
  type: boolean
repeat:
  ...
delay:
 ...
required: // are these required? if not, we should specify defaults
  - enabled
  - delay
  - repeat
```

Then the additional fields of low|high alerts can be derived from that using `allOf`. You can see an example in `./data/models/deviceevent/alarm.v1.yaml` (derives from `./data/models/deviceevent/event.v1.yaml`).

Reply:
I started down that path, but quickly hit the limits of my OpenAPI knowledge, and got a bit frustrated. There are five alerts, and at least three unique different alert configurations between them (maybe more, I don't recall off of the top of my head), but I'll take another stab at extracting some of the common bits.

Reply:
Feel free to pick my brains on OpenAPI. I've spent a fair bit of time wrapping my head around it 😬

Reply:
TL; DR - it is not always obvious or even intuitive, IMNSHO

Reply:
I've taken another stab at it (with some success even), let me know if that looks more how you'd expect it to.

I think I have a better grasp of where and when allOf/anyOf/oneOf can be used, and that's part of what was missing... That and its relationship to siblings. Repeated careful reading of [the docs](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/) paid off.